smoke-test/src/test/java/org/opennms/smoketest/dcb/DcbMultipleNodesIT.java

```diff
@@ -0,0 +1,220 @@
+        targetAddress = getContainerInternalIpAddress(TARGET_CONTAINER);
+
+        restClient = new RestClient(STACK.opennms().getWebAddress());
+
+        final Requisition requisition = new Requisition();
+        requisition.setForeignSource(FOREIGN_SOURCE);
+
+        // Time prep for the schedule
+        LocalTime time = LocalTime.now().plusMinutes(2);
+        String schedule = String.format("0 %d %d ? * *", time.getMinute(), time.getHour());
+
+        // Node in default location for triggering backup from core
+        // TODO make a loop for creation of at least 1000 nodes
+        for (int i = 1; i < 101; i++) {
+            final List<RequisitionMonitoredService> monitoredServiceList = new ArrayList<>();
+            monitoredServiceList.add(new RequisitionMonitoredService(DCB_SVC_NAME));
+
+            final RequisitionInterface requisitionInterface = new RequisitionInterface();
+            requisitionInterface.setIpAddr(targetAddress);
+            requisitionInterface.setManaged(true);
+            requisitionInterface.setSnmpPrimary(PrimaryType.PRIMARY);
+            requisitionInterface.setMonitoredServices(monitoredServiceList);
+
+            final List<RequisitionMetaData> metaDataList = new ArrayList<>();
+            metaDataList.add(new RequisitionMetaData("requisition", "dcb:config-type", DCB_CONFIG_TYPE));
+            metaDataList.add(new RequisitionMetaData("requisition", "dcb:username", DCB_USERNAME));
+            metaDataList.add(new RequisitionMetaData("requisition", "dcb:password", DCB_PASSWORD));
+            metaDataList.add(new RequisitionMetaData("requisition", "dcb:script-file", DCB_SCRIPT_NAME));
+            metaDataList.add(new RequisitionMetaData("requisition", "dcb:ssh-port", Integer.toString((SSH_PORT))));
+            metaDataList.add(new RequisitionMetaData("requisition", "dcb:schedule", schedule));
+
+            final RequisitionNode requisitionNode = new RequisitionNode();
+            requisitionNode.setForeignId("local_" + i);
+            requisitionNode.setNodeLabel("local_" + i);
+            requisitionNode.setLocation(null);
+            requisitionNode.setMetaData(metaDataList);
+            requisitionNode.putInterface(requisitionInterface);
+
+            requisition.insertNode(requisitionNode);
+        }
+
+        restClient.addOrReplaceRequisition(requisition);
+        restClient.importRequisition(FOREIGN_SOURCE);
+
+        final NodeDao nodeDao = STACK.postgres().dao(NodeDaoHibernate.class);
+        final IpInterfaceDao ipInterfaceDao = STACK.postgres().dao(IpInterfaceDaoHibernate.class);
+        final MonitoredServiceDao monitoredServiceDao = STACK.postgres().dao(MonitoredServiceDaoHibernate.class);
+
+        localNode = await()
+                .atMost(3, MINUTES)
+                .pollInterval(30, SECONDS)
+                .until(DaoUtils.findMatchingCallable(nodeDao, new CriteriaBuilder(OnmsNode.class).eq("foreignId", "local_1").toCriteria()), notNullValue());
```

Comment:
IMO, this should count for 100 nodes with the foreign source.

Reply:
Creation of nodes comes from the last to the first, whenever node 1 is in the list means all the rest are already there. It's more of a check for API work rather than a test. The real amount will be checked in the test.